### PR TITLE
Image enhacement button bug fix

### PR DESCRIPTION
**Pull request by 007quardo**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,10 +23,11 @@
     "log": "git log master --pretty=format:\\\"%an\\\",\\\"%ai\\\",\\\"%s\\\",\\\"%H\\\" > log.csv && echo Commiter, Date, Message, Hash | cat - log.csv > log2.csv && mv log2.csv log.csv",
     "ng": "ng",
     "prestart": "node scripts/generate-config.js",
-    "start": "ng serve",
+    "start": "node --max_old_space_size=12288 ./node_modules/@angular/cli/bin/ng serve",
     "start_1": "npm start -- --port 4201",
     "start:aot": "npm start -- --aot true",
     "start:prod": "npm start -- --prod true",
+    "start:max-space": "node --max_old_space_size=12288 ./node_modules/@angular/cli/bin/ng serve",
```
Comment (tltk90) [inline package.json:30]: why we need this script?
all the start should be run with -max-old-space.

File: package.json [hunk 2]
```diff
@@ -86,7 +87,7 @@
     "zone.js": "~0.10.2"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.1000.4",
+    "@angular-devkit/build-angular": "^0.1100.7",
```
Comment (tltk90) [inline package.json:90]: **revert**
as we are on angular 10.

File: src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts [hunk 3]
```diff
@@ -189,8 +192,19 @@ export class OverlayStatusComponent implements OnInit, OnDestroy, IEntryComponen
 		if (this.isDragged) {
 			this.toggleDragged();
 		}
+
+		this.onClickOutSide$.unsubscribe();
 	}
 
+	ngAfterViewInit() {
```
Comment (tltk90) [inline src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts:199]: maybe we can change `Autosubscriptions` init function to `ngAfterViewInit`

File: angular.json [hunk 4]
```diff
@@ -605,5 +603,8 @@
     "@schematics/angular:directive": {
       "prefix": "ansyn"
     }
+  },
```
Comment (tltk90) [inline angular.json:606]: why we need this?

**Review by tltk90 — CHANGES_REQUESTED** [review]
remove the report files

File: report.20210120.120429.1636.0.001.json [hunk 5]
```diff
@@ -0,0 +1,693 @@
+
```
Comment (ramtob) [inline report.20210120.120429.1636.0.001.json:1]: Unrelated file - please remove

File: report.20210120.120751.15084.0.001.json [hunk 6]
```diff
@@ -0,0 +1,699 @@
+
```
Comment (ramtob) [inline report.20210120.120751.15084.0.001.json:1]: Unrelated file - please remove

File: src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts [hunk 7]
```diff
@@ -92,14 +103,6 @@ export class OverlayStatusComponent implements OnInit, OnDestroy, IEntryComponen
 		})
 	);
 
-	@AutoSubscription
```
Comment (ramtob) [inline src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts:95]: Do not delete. AutoSubscription is great, and this is not the way to deal with it

File: src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts (hunk 3 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts:199]: @tltk90 excellent idea

@007quardo please do not remove AutoSubscription. Even if we can't do what @tltk90 suggested, we get just raise a flag in ngAfterViewInit(), and then check the flag in the observable in AutoSubscription

**Comment by tltk90** [comment]
@ramtob @007quardo 
we could also pass `autosubscriptions` the `ngAfterViewInit` function as the init function,

**Comment by 007quardo** [comment]
@tltk90 @ramtob 
I don't think there is a way, because the AutoSubs init is like a singleton for the whole class . I didn't see a way creating another instance of it , or using the same instance with other parameters.

**Comment by tltk90** [comment]
@007quardo the `autoSubscriptions` decorator of the class can get `options` object that could pass the init function and the destroy function( that was default to `ngOnInit` and `ngOnDestroy`).
if we change the init to `ngAfterViewInit`  the `element` will be defined.
you can see [here](https://github.com/yairtawil/auto-subscriptions/blob/master/src/auto-subscriptions.ts#L72) for more details.

**Comment by ramtob** [comment]
@tltk90 What @007quardo is perhaps meaning, is that the class contains more than one AutoSubscription decorators. So if you change the parameters of the AutoSubscriptions decorator, it will affect all the them.

However:
1. Not sure it will cause any problem, if all the AutoSubscrption's will subscribe on onAfterViewInit()
2. And even if it will, there is the other way that I suggested, to raise a flag in ngAfterViewInit(), and to check the flag where it is relevant.

File: src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts [hunk 8]
```diff
@@ -42,9 +57,10 @@ import { ComponentVisibilityItems } from '../../../app-providers/component-mode'
 	init: 'ngOnInit',
 	destroy: 'ngOnDestroy'
 })
-export class OverlayStatusComponent implements OnInit, OnDestroy, IEntryComponent {
+export class OverlayStatusComponent implements OnInit, OnDestroy, IEntryComponent, AfterViewInit {
 	// for component
 	readonly isAnnotationsShow: boolean;
+	@ViewChild('closeElementAnnotation', {static: false}) closeElementAnnotation: ElementRef;
```
Comment (tltk90) [inline src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts:63]: move after the `for component` section.

**Review by tltk90 — CHANGES_REQUESTED** [review]

File: src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts [hunk 9]
```diff
@@ -168,6 +169,14 @@ export class OverlayStatusComponent implements OnInit, OnDestroy, IEntryComponen
 			})
 		);
 
+	@AutoSubscription
+	onClickOutSide = () => this.isAfterViewInit.pipe(switchMap(() => this.clickOutsideService.onClickOutside({monitor: this.closeElementAnnotation.nativeElement}).pipe(
```
Comment (ramtob) [inline src/app/@ansyn/ansyn/modules/overlays/overlay-status/overlay-status.component.ts:173]: Very good.

I don't think we need two pipe()?
This is not considered good practice.
Why not with one pipe(), like:

pipe(
  switchMap(...),
  filter(..)
  ...
}?

**Review by ramtob — CHANGES_REQUESTED** [review]

**Review by ramtob — APPROVED** [review]
🥇

**Review by tltk90 — APPROVED** [review]
